natsstreaming/nats_streaming.go

```diff
@@ -205,6 +205,7 @@ func handleAcks(ctx context.Context, msgsToAck chan *consumerMessage, acks <-cha
 			if err := msgToAck.m.Ack(); err != nil {
 				return fmt.Errorf("failed to ack message with NATS: %v", err.Error())
 			}
+			toAck = toAck[1:]
```

Comment:
Potential memory leak because the underlying array has pointer data structs?

Reply:
Will write some tests to check, apart from that looks ok

Reply:
@vnadgir did you check what you wanted? Are you happy with this to go in?  Maybe if there's more work to do, we can create an issue for that?